libs/components/src/color-password/color-password.component.ts

```diff
@@ -14,19 +11,24 @@ enum CharacterType {
 
 @Component({
   selector: "bit-color-password",
-  template: `@for (character of passwordArray; track character; let i = $index) {
+  template: `@for (character of passwordCharArray(); track $index; let i = $index) {
     <span [class]="getCharacterClass(character)">
       <span>{{ character }}</span>
-      @if (showCount) {
+      @if (showCount()) {
         <span class="tw-whitespace-nowrap tw-text-xs tw-leading-5 tw-text-main">{{ i + 1 }}</span>
       }
     </span>
   }`,
   standalone: true,
 })
 export class ColorPasswordComponent {
-  @Input() password: string = null;
-  @Input() showCount = false;
+  password = input<string>("");
+  showCount = input<boolean>(false);
+
+  passwordCharArray = computed(() => {
+    // // Convert to an array to handle cases that strings have special characters, i.e.: emoji.
+    return (this.password() ?? []) ? Array.from(this.password()) : [];
+  });
```

Comment:
Double comment. We can also put it outside which makes it slightly more clear.
```suggestion
  // Convert to an array to handle cases that strings have special characters, i.e.: emoji.
  passwordCharArray = computed(() => {
    return (this.password() ?? []) ? Array.from(this.password()) : [];
  });
```